zim/plugins/linkmap.py

```diff
@@ -157,13 +176,39 @@ def __init__(self, parent, linkmap, navigation):
 			(Gtk.STOCK_ZOOM_OUT, self.xdotview.on_zoom_out),
 			(Gtk.STOCK_ZOOM_FIT, self.xdotview.on_zoom_fit),
 			(Gtk.STOCK_ZOOM_100, self.xdotview.on_zoom_100),
+			(Gtk.STOCK_REFRESH, self.refresh_xdotview),
 		):
 			button = IconButton(stock)
 			button.connect('clicked', method)
 			vbox.pack_start(button, False, True, 0)
 
+	def set_dotcode(self, page):
+		logger.debug('Load dotcode for page %s', page.name)
+		self.set_title(page.name + self.title_ending)
+		self.xdotview.set_dotcode(
+			LinkMap(self.pageview.notebook, page).get_dotcode().encode('UTF-8'))
+		if self.preferences['zoom_fit']:
+			self.xdotview.on_zoom_fit(0) # takes an action, doesn't use it
+
+	def refresh_xdotview(self, *a):
+		if self.preferences['refresh_follows_parent']:
+			self.set_dotcode(self.pageview.page)
+		else:
+			self.set_dotcode(self.page)
+
 	def on_node_clicked(self, widget, name, event):
 		if re.match('b\'.*?\'$', name):
 			# Bug in dotcode ? URLS come in as strings containing byte representation
 			name = ast.literal_eval(name).decode('UTF-8')
-		self.navigation.open_page(Path(name))
+		
+		# new_pageview is either our parent or a new PageView inside a new window
+		# disregard that; NavigationModel returns the old window's pageview
```

Comment:
NavigationModel's open_page returns the same pageview regardless of the new_window argument. Is this by design?